packages/framework/src/Model/Product/ProductOnCurrentDomainFacade.php

```diff
@@ -51,154 +49,27 @@ public function getVisibleProductById(int $productId): Product
-
-        return $this->productRepository->getPaginationResultForListableForBrand(
-            $brand,
-            $this->domain->getId(),
-            $this->domain->getLocale(),
-            $orderingModeId,
-            $this->currentCustomerUser->getPricingGroup(),
-            $page,
-            $limit,
-        );
-    }
-
-    /**
-     * @param string $searchText
-     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $productFilterData
-     * @param string $orderingModeId
-     * @param int $page
-     * @param int $limit
-     * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
-     */
-    public function getPaginatedProductsForSearch(
-        string $searchText,
-        ProductFilterData $productFilterData,
-        string $orderingModeId,
-        int $page,
-        int $limit,
-    ): PaginationResult {
-        return $this->productRepository->getPaginationResultForSearchListable(
-            $searchText,
-            $this->domain->getId(),
-            $this->domain->getLocale(),
-            $productFilterData,
-            $orderingModeId,
-            $this->currentCustomerUser->getPricingGroup(),
-            $page,
-            $limit,
-        );
-    }
-
-    /**
-     * @param string|null $searchText
-     * @param int $limit
-     * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
-     */
-    public function getSearchAutocompleteProducts(?string $searchText, int $limit): PaginationResult
-    {
-        $emptyProductFilterData = new ProductFilterData();
-
-        $page = 1;
-
-        return $this->productRepository->getPaginationResultForSearchListable(
-            $searchText,
-            $this->domain->getId(),
-            $this->domain->getLocale(),
-            $emptyProductFilterData,
-            ProductListOrderingConfig::ORDER_BY_RELEVANCE,
-            $this->currentCustomerUser->getPricingGroup(),
-            $page,
-            $limit,
-        );
-    }
-
     /**
      * @param int $categoryId
      * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfig $productFilterConfig
      * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $productFilterData
+     * @param string $searchText
      * @return \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterCountData
      */
     public function getProductFilterCountDataInCategory(
         int $categoryId,
         ProductFilterConfig $productFilterConfig,
         ProductFilterData $productFilterData,
+        string $searchText = '',
     ): ProductFilterCountData {
         $productsQueryBuilder = $this->productRepository->getListableInCategoryQueryBuilder(
             $this->domain->getId(),
             $this->currentCustomerUser->getPricingGroup(),
             $this->categoryRepository->getById($categoryId),
         );
 
+        // database filtering does not support limit by search
+
```

Comment:
This makes me think - is the implementation for Postgres still needed? Should we remove it too? I vote for removing.

Reply:
removed :)